### DISCUSSION TITLE
Moving Components with LUA

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#268] [Show and tell]
I couldn't find how to do this on the old forums

If you need to move a component with LUA this code works.

```
mainText = panel:getModulatorByName("mainText")

mainText:getComponent():setTopLeftPosition(193, 35)
```

**Comment by dobo365** [comment]
You can get all properties of a component with
`what(panel:getModulatorByName("modulator-2"):getComponent())`
for example

I started documenting them but I need to compare with one file that is in the source code.
This is one thing I wanted to improve by recompiling the 5.3.201: getting an help popup or mouse over bubble with help text.
For the moment we have nice links but they don't show anything...

**Reply by Godlike-Productions** [reply]
Yeah, not all of the items displayed with the what command necessarily do anything. You get back a list of operations from Juce, I think, but not all of them are necessarily mapped to ctrlr.

**Comment by shooking** [comment]
I used a similar trick when creating the K1r panel. So I had 32 LCDs. I gave them a unique ID (custom setting) then used the lua debugger to position them. I am no artist but I can do math.

**Reply by Godlike-Productions** [reply]
Oh cool. Wish I'd known where to look rather than hacking away at the console for a bit.

I'll check that panel out.

**Comment by shooking** [comment]
I found the link where I posted originally

https://ctrlr.org/forums/topic/using-lua-to-layout-stuff-repost/

**Reply by Godlike-Productions** [reply]
Yeah, I did come across that and initially thought it had methods only related to a rectangle object, but see now I probably could have used it for moving the image around.  setTopLeftPosition will always keep the object the same size though, and is simpler in that case. If you need to resize then the method you posted is better.
